src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/writers/XHTMLTemplateWriter.java

```diff
@@ -4,28 +4,42 @@
  */
 package org.geoserver.featurestemplating.writers;
 
+import static org.geoserver.featurestemplating.builders.VendorOptions.JSON_LD_SCRIPT;
 import static org.geoserver.featurestemplating.builders.VendorOptions.LINK;
 import static org.geoserver.featurestemplating.builders.VendorOptions.SCRIPT;
 import static org.geoserver.featurestemplating.builders.VendorOptions.STYLE;
 
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.OutputStream;
 import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.util.Collections;
 import java.util.List;
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
 import javax.xml.namespace.QName;
 import javax.xml.stream.XMLStreamException;
 import javax.xml.stream.XMLStreamWriter;
 import javax.xml.stream.events.Attribute;
 import org.geoserver.featurestemplating.builders.EncodingHints;
+import org.geotools.feature.FeatureCollection;
 import org.geotools.geometry.jts.ReferencedEnvelope;
 import org.geotools.geometry.jts.WKTWriter2;
 import org.locationtech.jts.geom.Geometry;
 
 /** A template writer able to produce XHTML output. */
 public class XHTMLTemplateWriter extends XMLTemplateWriter {
 
-    public XHTMLTemplateWriter(XMLStreamWriter streamWriter) {
+    private OutputStream output;
```

Comment:
Is this field being used? I don't see it.

Reply:
now it is used :)